### PR TITLE
Port account_invoice_supplier_ref_unique to v8 and new API

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

**Comment by ameyioli** [comment]
:+1:

**Comment by max3903** [comment]
Tested on runbot
:+1:

**Comment by cdubuit** [comment]
Useful :+1:

File: account_invoice_supplier_ref_unique/README.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Unique Supplier Invoice Number
+==============================
+
+This module checks that a supplier invoice/refund is not entered twice. This is important because if you enter twice the same supplier invoice, there is also a risk that you pay it twice !
+
+This module adds a constraint on supplier invoice/refunds to check that (commercial_partner_id, supplier_invoice_number) is unique, without considering the case of the supplier invoice number.
+
```
Comment (eLBati) [inline account_invoice_supplier_ref_unique/README.rst:7]: Please add 'bug tracker' section, like https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: account_invoice_supplier_ref_unique/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    @api.constrains('supplier_invoice_number')
+    def _check_unique_supplier_invoice_number_insensitive(self):
+        if (
+                self.supplier_invoice_number and
+                self.type in ('in_invoice', 'in_refund')):
+            same_supplier_inv_num = self.search([
+                ('commercial_partner_id', '=', self.commercial_partner_id.id),
+                ('type', 'in', ('in_invoice', 'in_refund')),
+                ('supplier_invoice_number',
+                 'ilike',
```
Comment (eLBati) [inline account_invoice_supplier_ref_unique/account_invoice.py]: I would use `=ilike` here, otherwise `123` will match with `12345`

**Pull request merged**